File: packages/python/examples/01_timeline.py
"""認証してタイムラインを取得する最小例。

    YAY_EMAIL=... YAY_PASSWORD=... python examples/01_timeline.py
"""

import asyncio
import os

import yaylib


async def main() -> None:
    # `async with` で基盤接続が確実にクローズされます。
    async with yaylib.Client() as client:
        await client.login_with_email(
            email=os.environ["YAY_EMAIL"],
            password=os.environ["YAY_PASSWORD"],
        )

        timeline = await client.get_timeline(
            noreply_mode=yaylib.NoreplyMode.EMPTY, number=20
        )
        for post in timeline.posts or []:
            print(post.id, post.text)


asyncio.run(main())

File: examples/typescript/01_timeline.ts
// 認証してタイムラインを取得する最小例。
//
//   YAY_EMAIL=... YAY_PASSWORD=... npx tsx examples/typescript/01_timeline.ts
import { Client, NoreplyMode } from "yaylib";

const client = new Client();

await client.loginWithEmail({
  email: process.env.YAY_EMAIL!,
  password: process.env.YAY_PASSWORD!,
});

const timeline = await client.getTimeline({
  noreplyMode: NoreplyMode.False,
  number: 20,
});

for (const post of timeline.posts ?? []) {
  console.log(post.id, post.text);
}
